storage/gcloud/aio/storage/storage.py

```diff
@@ -115,9 +126,113 @@ def __init__(self, project, service_file, token=None, session=None):
         resp.raise_for_status()
         return await resp.json()
 
-    async def download_as_string(self, bucket, object_name, session=None):
-        return await self.download(bucket, object_name,
-                                   params={'alt': 'media'}, session=session)
+    async def _upload_resumable(self, url: str, bucket: str, object_name: str,
+                                data, headers: dict = None,
+                                session: aiohttp.ClientSession = None, timeout: int = 120,
+                                upload_type_resumable_always: bool = False):
+        # https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload
+        async def initiate_upload_session(object_name: str, headers: dict, session: aiohttp.ClientSession):
+            params = {
+                'uploadType': 'resumable',
+            }
+
+            metadata = {'name': object_name}
+            metadata_str = json.dumps(metadata)
+            metadata_len = len(metadata_str)
+
+            post_headers = {**headers}
+            post_headers.update({
+                'Content-Length': str(metadata_len),
+                'Content-Type': 'application/json; charset=UTF-8',
+                'X-Upload-Content-Type': headers['Content-Type'],
+                'X-Upload-Content-Length': headers['Content-Length']
+            })
+
+            resp = await session.post(url, headers=post_headers, params=params, data=metadata_str, timeout=5)
+            resp.raise_for_status()
+
+            header_with_session_URI = dict(resp.headers)
```

Comment:
I think everything from this line down can just be replaced with `return resp.headers['Localtion']`, since the resulting `KeyError` if the location URI is missing would be clear enough.